backend/drm/drm.c

```diff
@@ -88,118 +164,68 @@ static bool init_planes(struct wlr_drm_backend *drm) {
 
 	wlr_log(WLR_INFO, "Found %"PRIu32" DRM planes", plane_res->count_planes);
 
-	if (plane_res->count_planes == 0) {
-		drmModeFreePlaneResources(plane_res);
-		return true;
-	}
-
-	drm->num_planes = plane_res->count_planes;
-	drm->planes = calloc(drm->num_planes, sizeof(*drm->planes));
-	if (!drm->planes) {
-		wlr_log_errno(WLR_ERROR, "Allocation failed");
-		goto error_res;
-	}
-
-	for (size_t i = 0; i < drm->num_planes; ++i) {
-		struct wlr_drm_plane *p = &drm->planes[i];
+	for (uint32_t i = 0; i < plane_res->count_planes; ++i) {
+		uint32_t id = plane_res->planes[i];
 
-		drmModePlane *plane = drmModeGetPlane(drm->fd, plane_res->planes[i]);
+		drmModePlane *plane = drmModeGetPlane(drm->fd, id);
 		if (!plane) {
 			wlr_log_errno(WLR_ERROR, "Failed to get DRM plane");
-			goto error_planes;
+			goto error;
 		}
 
-		p->id = plane->plane_id;
-		p->possible_crtcs = plane->possible_crtcs;
+		union wlr_drm_plane_props props = {0};
+		if (!get_drm_plane_props(drm->fd, id, &props)) {
+			drmModeFreePlane(plane);
+			goto error;
+		}
 
 		uint64_t type;
-		if (!get_drm_plane_props(drm->fd, p->id, &p->props) ||
-				!get_drm_prop(drm->fd, p->id, p->props.type, &type)) {
+		if (!get_drm_prop(drm->fd, id, props.type, &type)) {
 			drmModeFreePlane(plane);
-			goto error_planes;
+			goto error;
 		}
 
-		p->type = type;
-		drm->num_type_planes[type]++;
-
-		// Choose an RGB format for the plane
-		uint32_t rgb_format = DRM_FORMAT_INVALID;
-		for (size_t j = 0; j < plane->count_formats; ++j) {
-			uint32_t fmt = plane->formats[j];
-			wlr_drm_format_set_add(&p->formats, fmt, DRM_FORMAT_MOD_INVALID);
-
-			if (fmt == DRM_FORMAT_ARGB8888) {
-				// Prefer formats with alpha channel
-				rgb_format = fmt;
-				break;
-			} else if (fmt == DRM_FORMAT_XRGB8888) {
-				rgb_format = fmt;
+		/*
+		 * This is a very naive implementation of the plane matching
+		 * logic. Primary and cursor planes should only work on a
+		 * single CRTC, and this should be perfectly adequate, but
+		 * overlay planes can potentially work with multiple CRTCs,
+		 * meaning this could return inefficent/skewed results.
+		 *
+		 * However, we don't really care about overlay planes, as we
+		 * don't support them yet. We only bother to keep basic
+		 * tracking of them from DRM lease clients.
+		 */
+		int crtc_i = ffs(plane->possible_crtcs);
```

Comment:
Can we rename this to `crtc_pos` instead of `crtc_i`? It starts at 1, so using "index" is confusing.

Reply:
Makes sense. I went with `crtc_bit` and added more clarification as to what it means in the comment.